packages/tests/server/websockets.test.ts

```diff
@@ -1656,7 +1658,10 @@ describe('keep alive from the client', () => {
 
     expect(pong).toBe(false);
 
-    expect(onClose).toHaveBeenCalledTimes(1);
+    await vi.useRealTimers();
+    await waitFor(() => {
+      expect(onClose).toHaveBeenCalledTimes(1);
+    });
```

Comment:
Some async/await calls were added to the open & close flow, so we now need to wait for this event.

We could easily revert this change by adding a reference to `wsClient.opts.callbacks.onClose` at the ping/pong level (so that ping/pong can call it synchronously). However, this breaks the separation of responsibilities, as a low-level component would then need to hold a reference to a high-level method. In my opinion, it’s not worth it.